src/ProjectTemplates/Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Pages/Index.cshtml.cs

```diff
@@ -2,16 +2,64 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+#if (GenerateApiOrGraph)
+using Microsoft.Extensions.Configuration;
+using Microsoft.Identity.Web;
+using System.Net;
+using System.Net.Http;
+using Company.WebApplication1.Services;
+#endif
+#if (CallsMicrosoftGraph)
+using Microsoft.Graph;
+#endif
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.AspNetCore.Mvc.RazorPages;
 using Microsoft.Extensions.Logging;
 
 namespace Company.WebApplication1.Pages
 {
+#if (GenerateApiOrGraph)
+    [AuthorizeForScopes(ScopeKeySection = "CalledApi:CalledApiScopes")]
+#endif 
     public class IndexModel : PageModel
     {
         private readonly ILogger<IndexModel> _logger;
 
+#if (GenerateApi)
```

Comment:
Looking at this logic, I think we will need more validation. For example, what should happen if the user specifies noAuth but specifies CalledApiUrl? Does using this block of code make sense in that scenario?

Reply:
you are right. I'll fix it.
Raised https://github.com/AzureAD/microsoft-identity-web/issues/337 for this
cc: @henrik-me @jennyf19 @pmaytak

Reply:
I'm going to change the templates to only compute GenerateApi and CallsMicrosoftGraph as true when either IndividualB2CAuth or OrganizationalAuth is specified. Otherwise, even if the options are specified, they will be ignored.

Reply:
I added validation in https://github.com/dotnet/aspnetcore/pull/24065/commits/3644873b74c30fa673915e3ce8cc496495699af2